src/lib/snarky_bowe_gabizon_hash/snarky_bowe_gabizon_hash.ml

```diff
@@ -0,0 +1,49 @@
+open Core_kernel
+open Fold_lib
+open Snarky
+
+(* TODO: Have a compatibility layer so we only need the module M. *)
+module Make
+    (M : Snark_intf.Run)
+    (Impl : Snark_intf.S with type field = M.field)
+    (Inputs : Inputs_intf.S
+              with type field := M.field
+               and type ('a, 'b) checked := ('a, 'b) Impl.Checked.t) =
+struct
+  open Inputs
+  open M
+
+  let bottom_bit y = List.hd_exn (Field.unpack_full y :> Boolean.var list)
+
+  let choose_unpacking = Field.choose_preimage_var ~length:Field.size_in_bits
+
+  let g1_to_bits (x, y) = bottom_bit y :: choose_unpacking x
```

Comment:
should we have these as functions defined in a more central place so that they can be reused (without bringing this in as a dependency)?

Reply:
I couldn't find a great place for this right now. Is it ok if we make it a TODO as this is in the critical path of the BG changeover?